fgpyo/sam/__init__.py

```diff
@@ -1164,6 +1236,31 @@ def all_recs(self) -> Iterator[AlignedSegment]:
             for rec in recs:
                 yield rec
 
+    def set_mate_info(
+        self,
+        is_proper_pair: Callable[[AlignedSegment, AlignedSegment], bool] = is_proper_pair,
+        isize: Callable[[AlignedSegment, AlignedSegment], int] = isize,
+    ) -> Self:
+        """Reset all mate information on every alignment in the template.
+
+        Args:
+            is_proper_pair: A function that takes two alignments and determines proper pair status.
+            isize: A function that takes the two alignments and calculates their isize.
+        """
+        if self.r1 is not None and self.r2 is not None:
+            set_mate_info(self.r1, self.r2, is_proper_pair=is_proper_pair, isize=isize)
+        if self.r1 is not None:
+            for rec in self.r2_secondaries:
+                set_mate_info_on_secondary(secondary=rec, mate_primary=self.r1)
+            for rec in self.r2_supplementals:
+                set_mate_info_on_supplementary(supp=rec, mate_primary=self.r1)
+        if self.r2 is not None:
+            for rec in self.r1_secondaries:
+                set_mate_info_on_secondary(secondary=rec, mate_primary=self.r2)
+            for rec in self.r1_supplementals:
+                set_mate_info_on_supplementary(supp=rec, mate_primary=self.r2)
+        return self
```

Comment:
I was expecting this:
```suggestion
        if self.r1 is not None:
            for rec in self.all_r2s:
                _set_common_mate_fields(dest=rec, source=self.r1)
        if self.r2 is not None:
            for rec in self.all_r1s:
                _set_common_mate_fields(dest=rec, source=self.r2)
        return self
```

Reply:
With my last commit, I still have two separate functions for secondaries and supplementaries. They aren't that different from `_set_common_mate_fields` except for:

- Additional checking for if the record is secondary or supplementary for safety sake
- Proper pair and template length settings on supplementaries (but only for non-secondary supplementaries).